Changelog 200 (#188)
intellij-rust/intellij-rust.github.io: _posts/2023-08-07-changelog-200.markdown

---
layout: post
title: "IntelliJ Rust Changelog #200"
date: 2023-08-07 13:00:00 +0300
---

## New Features

* Add [Code Vision hints](https://www.jetbrains.com/help/rider/Code_Vision.html)

  * [#10763] Usages hints for some top-level items

  * [#10756] VCS hints

>   * [#9958] Implementations (inheritors) hints. Note, these hints are available starting with 2022.3 and are currently disabled by default. You can turn it on via `Registry... | org.rust.code.vision.implementation` (by [@Kobzol])

Reviewer: Let's move it to the `internal` section since it's disabled by default

  <img src="/assets/posts/changelog-200/code_vision.png" width="700px"/>

* [#10744] Improve completion of `if` and `match` keywords

  {% include gif-img.html path="/assets/posts/changelog-200/match" w="350px" %}

> * [#10718] Add macro expansion to quick documentation on hover

Reviewer 2: ```suggestion
* [#10718] Show the macro expansion in quick documentation on hover
```

> * [#10729] Add quick fix for printing struct which doesn't implement `Debug` or `Display`

Reviewer 2: ```suggestion
* [#10729] Add the quick fix for printing a struct that doesn't implement `Debug` or `Display` traits
```

> * [#9424] The intention to convert a match expression to a series of if let statements is now offered in more situations (by [@Kobzol])

Reviewer 2: ```suggestion
* [#9424] The intention to convert a `match` expression to a series of `if let` expressions is now offered in more situations (by [@Kobzol])
```

* [#10716] Respect `CARGO_HOME` env variable when looking for Rust toolchain

* [#10739] Support inspection "Unused must_use" for `impl Trait`

* [#10714] Support [`c_str_literal` feature](https://rust-lang.github.io/rfcs/3348-c-str-literal.html)

> * [#10726] Add inspection to replace `therad_rng().gen()` to `random()` from [`rand`](https://docs.rs/rand/latest/rand/) crate

Reviewer 2: ```suggestion
* [#10726] Add inspection to replace `therad_rng().gen()` with `random()` from [`rand`](https://docs.rs/rand/latest/rand/) crate
```

## Fixes

> * [#10760] Adjust intellisense file size limit for Rust files. This fixes expansion of macros from [`swc`](https://github.com/swc-project/swc)

Reviewer 2: ```suggestion
* [#10760] Adjust IntelliSense file size limit for Rust files. This fixes the expansion of macros from [`swc`](https://github.com/swc-project/swc)
```

> * [#10755] Implement `impl`s search for an alias of alias and for a use alias

Reviewer 2: ```suggestion
* [#10755] Implement `impl`s search for an alias of an alias and for an alias usage
```
